packages/edit-site/src/components/global-styles/variations/variation.js

```diff
@@ -38,10 +44,20 @@ export default function Variation( { variation, children } ) {
 	);
 
 	const selectVariation = () => {
-		setUserConfig( () => ( {
-			settings: variation.settings,
-			styles: variation.styles,
-		} ) );
+		setUserConfig( ( currentConfig ) => {
+			// Avoids overwriting certain paths when applying a theme variation.
+			for ( const path of PATHS_TO_AVOID_OVERWRITING ) {
```

Comment:
I don't understand why this needs to be a loop. Isn't it only really one path we want to avoid overwriting?

Reply:
It is _currently_ but that could change in the future. If we wanted to simplify this PR we could avoid the loop and make the "path" a single string and not an array of strings.

Reply:
I wonder if that would be better, if we don't have other cases that this need to happen for.